free.cabal

```diff
@@ -82,46 +73,22 @@ library
     GADTs
 
   build-depends:
-    base                 >= 4.5     && < 5,
+    base                 >= 4.9     && < 5,
     comonad              >= 5.0.8   && < 6,
-    containers           >= 0.3     && < 0.7,
+    containers           >= 0.5.7.1 && < 0.7,
     distributive         >= 0.5.2   && < 1,
     exceptions           >= 0.10.4  && < 0.11,
     indexed-traversable  >= 0.1.1   && < 0.2,
+    mtl                  >= 2.2.2   && < 2.4,
+    profunctors          >= 5.6.1   && < 6,
     semigroupoids        >= 5.3.5   && < 6,
     th-abstraction       >= 0.4.2.0 && < 0.5,
-    transformers         >= 0.3     && < 0.7,
+    transformers         >= 0.5     && < 0.7,
     transformers-base    >= 0.4.5.2 && < 0.5,
-    template-haskell     >= 2.7.0.0 && < 2.20
-
-  -- GHC-7.8 bundles transformers-0.3,
-  -- mtl-2.2.* requires transformers >=0.4
-  if impl(ghc >=7.10)
-    build-depends:
-      mtl               >= 2.2.2 && < 2.4
-  else
-    build-depends:
-      mtl               >= 2.1.3.1 && < 2.4
-
-  -- recent profunctors dropped support for GHCs older than 7.8
-  if impl(ghc >=7.8)
-    build-depends:
-      profunctors >= 5.6.1 && < 6
-  else
-    build-depends:
-      profunctors >= 5.2.2 && < 5.3
+    template-haskell     >= 2.11    && < 2.20
 
   if !impl(ghc >= 8.2)
-    build-depends: bifunctors >= 5.5.9 && < 6
```

Comment:
That's a change. Now it doesn't depend on `bifunctors` at all anymore?

Reply:
This was replaced with the `bifunctor-classes-compat` library:

https://github.com/ekmett/free/blob/8f8b5e8734ef3ba97278789debed4962c1ea8279/free.cabal#L91

This was recently split out of `bifunctors` to provide all of the compatibility shims for `Bifunctor`, `Bifoldable`, and `Bitraversable` without any of the other machinery in `bifunctors` (which incurs additional dependencies). The next `bifunctors` release will also depend on `bifunctor-classes-compat`. See ekmett/semigroupoids#130 for the discussion that originally motivated this.

Strictly speaking, I don't have to replace `bifunctors` with `bifunctor-classes-compat` in this particular spot, but I figured that we might as well while we're in the neighborhood.